test/gtest/ucp/ucp_test.h

```diff
@@ -17,51 +17,51 @@ struct ucp_test_param {
     std::vector<std::string>  transports;
 };
 
-
-/**
- * UCP test
- */
-class ucp_test : public ucs::test_base, public ::testing::TestWithParam<ucp_test_param> {
-
+class ucp_test_entity {
 public:
-    UCS_TEST_BASE_IMPL;
+    ucp_test_entity(const ucp_test_param& test_param, ucp_config_t* ucp_config);
 
-    ucp_test();
-    virtual ~ucp_test();
+    void connect(const ucp_test_entity* other);
 
-    ucp_config_t* m_ucp_config;
+    void flush_ep() const;
 
-    class entity {
-    public:
-        entity(const ucp_test_param& test_param, ucp_config_t* ucp_config);
+    void flush_worker() const;
 
-        void connect(const entity* other);
+    void fence() const;
 
-        void flush_ep() const;
+    void disconnect();
 
-        void flush_worker() const;
+    void destroy_worker();
 
-        void fence() const;
+    ucp_ep_h ep() const;
 
-        void disconnect();
+    ucp_worker_h worker() const;
 
-        void destroy_worker();
+    ucp_context_h ucph() const;
 
-        ucp_ep_h ep() const;
+    void progress();
 
-        ucp_worker_h worker() const;
+protected:
+    ucs::handle<ucp_context_h> m_ucph;
+    ucs::handle<ucp_worker_h>  m_worker;
+    ucs::handle<ucp_ep_h>      m_ep;
+};
 
-        ucp_context_h ucph() const;
+/**
+ * UCP test
+ */
+class ucp_test : public ucs::test_base,
+                 public ::testing::TestWithParam<ucp_test_param>,
+                 public ucs::entities_storage<ucp_test_entity> {
 
-        void progress();
+    friend class ucp_test_entity;
```

Comment:
@shssf we may keep entity as inner class (to avoid possible namespace collisions) this way:
```
class ucp_test_base : public ucs::test_base {
public:
 class entity {
 }
};

class ucp_test : public ucp_test_base, public ucs::entity_storage<ucp_test_base::entity> {
};
```